### PR TITLE
Updated YaST RPM macros

### PR DESCRIPTION
**Pull request by lslezak**
Added support for Rake based packages, optionally run additional tests (used at Jenkins CI builds).

- 3.1.39

### CODE REVIEW

File: build-tools/rpm/macros.yast [hunk 1]
```diff
@@ -67,13 +67,35 @@
...
             VERBOSE=1 \\\
             Y2DIR="%{buildroot}/%{yast_dir}" \\\
             DESTDIR="%{buildroot}" \
-    fi
+    elif [ -f "Rakefile" ]; then \
+        # TODO: coveralls support \
+        LC_ALL=en_US.utf8 rake --verbose --trace check:ci \
+    fi \
+    %else \
+    if [ ! -f "%{yast_ydatadir}/devtools/NO_MAKE_CHECK" ]; then \
```
Comment (mvidner) [inline build-tools/rpm/macros.yast]: I think we can remove this NO_MAKE_CHECK feature and move the `%if` inside the `if`

**Comment by mvidner** [comment]
The macros are getting quite ugly (several [*real* real backslashes](http://xkcd.com/1638/) FTW!) but I think it is still bearable now. If we need to make this any longer, we should separate the code to `build-tools/scripts`.

**Comment by lslezak** [comment]
Yes, I was also thinking about moving to separate scripts. IMHO it's at the edge but still acceptable... Let's see later, we will need to add some support for coveralls which will probably make the split a must.

File: build-tools/rpm/macros.yast [hunk 2]
```diff
@@ -67,13 +67,35 @@
...
+    if [ -f "configure.in.in" ]; then \
+        # TODO: fix the check:ci task to also work with autotools based modules \
         %{__make} check \\\
             VERBOSE=1 \\\
             Y2DIR="%{buildroot}/%{yast_dir}" \\\
             DESTDIR="%{buildroot}" \
-    fi
+    elif [ -f "Rakefile" ]; then \
+        # TODO: coveralls support \
+        LC_ALL=en_US.utf8 rake --verbose --trace check:ci \
```
Comment (mvidner) [inline build-tools/rpm/macros.yast]: It is better to use the canonical spelling, `en_US.UTF-8`

**Comment by mvidner** [comment]
LGTM

**Pull request merged**